### PR TITLE
Restrict hash used for computing generation to key hash

### PR DESCRIPTION
**Pull request by lucafavatella**
Depends on - and integrates - #1451

### CODE REVIEW

**Comment by lucafavatella** [comment]
Auxiliary [CI system tests job](https://circleci.com/gh/aeternity/epoch/26392) on 59b6d5a equivalent to 618cc981

**Comment by lucafavatella** [comment]
In [CI docker_smoke_tests job](https://circleci.com/gh/aeternity/epoch/26397) I see error:
```
#!/bin/bash -eo pipefail
sleep 30 # give it some time to boot the tcp listeners
EXTERNAL_ADDRESS=localhost:3001 INTERNAL_ADDRESS=localhost:3001/internal ./docker/healthcheck.sh
EXTERNAL_ADDRESS=localhost:3002 INTERNAL_ADDRESS=localhost:3002/internal ./docker/healthcheck.sh
EXTERNAL_ADDRESS=localhost:3003 INTERNAL_ADDRESS=localhost:3003/internal ./docker/healthcheck.sh

Exited with code 1
```

**Comment by lucafavatella** [comment]
Pushed fix.

**Comment by lucafavatella** [comment]
New auxiliary [CI system tests job](https://circleci.com/gh/aeternity/epoch/26413) on e666de9 equivalent to 547dfc2c

File: apps/aecore/src/aec_peer_connection.erl [hunk 1]
```diff
@@ -856,10 +856,30 @@ handle_get_generation_rsp(S, {get_generation, From, _TRef}, Msg) ->
     gen_server:reply(From, Res),
     remove_request_fld(S, get_generation).
 
+deserialize_key_block(SKB) ->
```
Comment (hanssv) [inline apps/aecore/src/aec_peer_connection.erl]: writing a parameterized `deserialize_block` would save some code duplication but introduce a wee bit more complexity... matter of taste really.

**Comment by lucafavatella** [comment]
> New auxiliary [CI system tests job](https://circleci.com/gh/aeternity/epoch/26413) on e666de9 equivalent to 547dfc2c

I see passed.

**Comment by lucafavatella** [comment]
Tested on dev1 environment from clean DB: ok - in sync at height 55

**Pull request merged**